build/includes/build-image.mk

```diff
@@ -23,15 +23,18 @@ build_remote_tag = $(REGISTRY)/$(build_tag)
 
 # Creates the build docker image
 build-build-image:
-	docker build --tag=$(build_tag) $(build_path)/build-image $(DOCKER_BUILD_ARGS)
+	docker build --tag=$(build_tag) $(build_path)/build-image \
+		--build-arg USER_ID="$(shell id -u ${USER})" \
```

Comment:
Would it work such that if `$(shell id -u ${USER})` was 0, then skip this step (and use the defaults)?

Could even do it as a top level Makefile items, such as:

```makefile
ifneq ($(shell id -u ${USER}), 0)
  docker_build_user := -build-arg USER_ID="$(shell id -u ${USER})" --build-arg GROUP_ID="$(shell id -g ${USER})"
endif
```

And then whenever building a docker image, use ${docker_build_user} in space of the details, and if it's root, it'll get ignored, which will be fine.

WDYT?

Reply:
That's a good idea (I was thinking something along those lines and doing it once in the Makefile way would make a lot of sense). 

One other idea that @zmerlynn suggested was to see if we could have cloud build run as a non-root user. But I think your suggestion might be simpler to implement (and test locally).

Reply:
> One other idea that @zmerlynn suggested was to see if we could have cloud build run as a non-root user. But I think your suggestion might be simpler to implement (and test locally).

I had a look to see if you could easily get cloud build to do that, and I couldn't find anything.

Reply:
I found one place that suggested that instead of using `gcr.io/cloud-builders/docker` you could create a new build container based off of that container and add a user to it. But this solution doesn't seem great because we'd have to keep updating a custom build image. 

The solution you proposed also adds a user (so the build runs as non-root), but doesn't require us to maintain our own base build container.

Reply:
Pushed a new commit with this suggestion. Let's see how it works. 🤞